### PR TITLE
Add support for Opensearch 2.3

### PR DESCRIPTION
**Pull request by viren-nadkarni**
### Summary

- Add support for Opensearch 2.3
- Make 2.3 the default version. The integration tests now run against this version
- Bumps opensearch-py to allow client support for OS 2.x. It is also mostly compatible with OS 1.x, functionality with OS 1.x won't break unless for things that are removed in 2.x [more info](https://github.com/opensearch-project/opensearch-py/blob/main/COMPATIBILITY.md)

### Notes

- Opensearch 2.3 has an updgrade path from Opensearch 1.3, [more info here](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/version-migration.html)
- AWS docs don't mention the default Opensearch version, but attempting to create a domain using CLI without a version specified causes Opensearch 2.3 to be used.

### Related
Docs: https://github.com/localstack/docs/pull/451

### CODE REVIEW

File: localstack/services/firehose/provider.py [hunk 1]
```diff
@@ -621,9 +620,7 @@ def _put_to_search_db(
                 )
             )
             try:
-                db_connection.create(
-                    index=search_db_index, doc_type=search_db_type, id=obj_id, body=body
-                )
+                db_connection.create(index=search_db_index, id=obj_id, body=body)
```
Comment (alexrashed) [inline localstack/services/firehose/provider.py:623]: q: Are you sure this is still compatible with older versions of OpenSearch?
According to the [compatibility docs you mentioned, Firehose does not yet support OpenSearch 2.3](https://docs.aws.amazon.com/opensearch-service/latest/developerguide/version-migration.html): 
> Kinesis Data Firehose currently doesn't support domains running version 2.3 as a destination. If you need to receive data from Kinesis Data Firehose, we recommend waiting until it's supported before you upgrade your domains to 2.3.

If this is not compatible, then I would propose the following steps:
- When determining the endpoint, also check if the version is compatible, and raise an error if it's not.
- Update the tests such that they test the error.
- Update the tests such that they use an older / compatible version of OpenSearch for the Firehose integration testing.

**Review by alexrashed — COMMENTED** [review]
Overall, the changes are looking great! I am only uncertain about the backwards compatibility of the changes in the firehose -> opensearch integration. As soon as this is clarified, we're good to go! 🚀

File: tests/integration/test_firehose.py [hunk 2]
```diff
@@ -126,7 +127,6 @@ def test_firehose_http(
 
 class TestFirehoseIntegration:
     @pytest.mark.skip_offline
-    @pytest.mark.parametrize("opensearch_endpoint_strategy", ["domain", "path"])
```
Comment (viren-nadkarni) [inline tests/integration/test_firehose.py:129]: Opensearch is not used in this test, so removed parametrisation

**Review by alexrashed — APPROVED** [review]
Awesome! Thanks for addressing the comments! Great to see the new major version of OpenSearch being supported in LocalStack! 🚀

**Pull request merged**